### ISSUE TITLE
weight clipping in RoBERTa

### ISSUE THREAD

**Issue by guolinke** [pytorch/fairseq#2532]
I loaded the released RoBERTa-large checkpoint, and find the `min(all parameters)` is exactly `-1.0`, and `max(all parameters)` is `1.0`. 
Does RoBERTa train by weight clipping to the range [-1, 1], or there are some post-processings after pretraining?

I cannot find the related descriptions from the RoBERTa paper and code.

**Comment by myleott** [comment]
Interesting. We didn't clip the weights during pretraining or postprocess the weights. The `roberta.base` model was trained in the exact same way, but weights are in [-1.0449, 1.3037].

**Comment by guolinke** [comment]
Thanks, @myleott , very interesting. 
I try to pre-train the roberta-large with 160G data from scratch. But I find the training is very unstable, the loss often explodes, and I need to decrease the learning rate.
Before explode, I find some weights become very large. So I check the weight of the released roberta-large, and find its range is [-1, 1]. This indeed confuses me.

**Comment by sh0416** [comment]
Any thought?
